bindinfo/handle.go

```diff
@@ -704,7 +704,11 @@ func GenerateBindSQL(ctx context.Context, stmtNode ast.StmtNode, planHint string
 		bindSQL = bindSQL[updateIdx:]
 		return strings.Replace(bindSQL, "UPDATE", fmt.Sprintf("UPDATE /*+ %s*/", planHint), 1)
 	case *ast.SelectStmt:
+		withIdx := strings.Index(bindSQL, "WITH")
 		selectIdx := strings.Index(bindSQL, "SELECT")
+		if withIdx != -1 && withIdx < selectIdx {
+			selectIdx = withIdx
+		}
 		// Remove possible `explain` prefix.
 		bindSQL = bindSQL[selectIdx:]
 		return strings.Replace(bindSQL, "SELECT", fmt.Sprintf("SELECT /*+ %s*/", planHint), 1)
```

Comment:
The hint would be added to the select keyword in the cte definition, seems like that it cannot take effect for the main query block.

Reply:
it will be effect on sql like `with cte as (select * from t where b=7) select * from cte left join t1 on cte.a=t1.id;`

Reply:
@zhaoxugang For example, `with cte as (select * from t) select * from cte, t1 use index(a)` will be restored to `with cte as select /*+ use_index(t1, a) */ * from t  select * from cte, t1`. It's wrong

Reply:
@wjhuang2016 I try this case,but got  the result:
+-----------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+--------+-------------------------+-------------------------+---------+-----------+---------+
| Original_sql                                                                            | Bind_sql                                                                                                                                                   | Default_db | Status | Create_time             | Update_time             | Charset | Collation | Source  |
+-----------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+--------+-------------------------+-------------------------+---------+-----------+---------+
| with `cte` as ( select * from `test` . `t` ) select * from ( `cte` ) join `test` . `t1` | WITH `cte` AS (SELECT /*+ use_index(@`sel_1` `test`.`t1` `a`), hash_join(@`sel_1` `test`.`t1`)*/ * FROM `test`.`t`) SELECT * FROM (`cte`) JOIN `test`.`t1` | test       | using  | 2021-07-10 15:04:27.388 | 2021-07-10 15:04:27.388 |         |           | capture |
+-----------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------+------------+--------+-------------------------+-------------------------+---------+-----------+---------+

Reply:
@zhaoxugang No, the hint is wrong. It doesn't work for the main select.

Reply:
> @zhaoxugang No, the hint is wrong. It doesn't work for the main select.

done